### PR TITLE
Expanding plugin error reporting from plugin to starter

### PR DESCRIPTION
**Pull request by joereuss12**
This is a work in progress still but current changes include:
- Populating the new `DeveloperData` classad created by Todd M with most of the fields specified in the document of the issue
- Changing some inner-workings of the client, so that we can pass all relevant information back to the plugin
- Classads package can now decipher types of map[string]interface{} for nested classAds

Work still needed:
- Still need to add a few more classads with more info on each attempt
- More testing by hand is needed, to ensure functionality and backwards compatibility
- Testing with Todd M's latest merge to HTCondor to ensure functionality there
- Need to figure out how/what to return for classads on uploads

### CODE REVIEW

File: client/handle_http.go [hunk 1]
```diff
@@ -264,15 +275,14 @@ func GenerateTransferDetailsUsingCache(cache CacheInterface, opts TransferDetail
 	return nil
 }
 
-func download_http(sourceUrl *url.URL, destination string, payload *payloadStruct, namespace namespaces.Namespace, recursive bool, tokenName string) (bytesTransferred int64, err error) {
-
+// func download_http(sourceUrl *url.URL, destination string, payload *payloadStruct, namespace namespaces.Namespace, recursive bool, tokenName string) (bytesTransferred int64, err error, attempts []Attempt) {
```
Comment (joereuss12) [inline client/handle_http.go:278]: Missed this on accident, it is removed

**Comment by joereuss12** [comment]
Currently not able to work on a submit AP however, I am opening this up for review

**Comment by joereuss12** [comment]
Just fixed merge conflicts

**Review by jhiemstrawisc — APPROVED** [review]
LGTM. Let's wait for the upstream build dependency to get fixed up and when the test can pass, I'll merge.

**Pull request merged**